src/org/jitsi/impl/neomedia/rtp/TransportCCEngine.java

```diff
@@ -330,41 +325,42 @@ public void tccReceived(RTCPTCCPacket tccPacket)
             }
         }
 
-        long referenceTime = tccPacket.getReferenceTime();
-        if (lastTimestampUs == -1)
+        RTCPTCCPacket.PacketMap packetMap = tccPacket.getPackets();
+        long previousArrivalTimeMs = -1;
+        for (Map.Entry<Integer, Long> entry : packetMap.entrySet())
         {
-            currentOffsetMs = System.currentTimeMillis();
-        }
-        else
-        {
-            long delta = referenceTime - lastTimestampUs;
-
-            // Detect and compensate for wrap-arounds in base time.
-            if (Math.abs(delta - kBaseTimestampRangeSizeUs) < Math.abs(delta))
+            long arrivalTimeUs = entry.getValue();
+            if (arrivalTimeUs == -1)
             {
-                delta -= kBaseTimestampRangeSizeUs;  // Wrap backwards.
+                continue;
             }
-            else if (Math.abs(delta + kBaseTimestampRangeSizeUs) < Math.abs(delta))
+
+            PacketDetail packetDetail = null;
+            synchronized (sentPacketsSyncRoot)
             {
-                delta += kBaseTimestampRangeSizeUs;  // Wrap forwards.
+                packetDetail = sentPacketDetails.remove(entry.getKey());
             }
 
-            currentOffsetMs += delta / 1000;
-        }
-
-        lastTimestampUs = referenceTime;
-
-        for (Map.Entry<Integer, Long> entry
-            : tccPacket.getPackets().entrySet())
-        {
-            PacketDetail packetDetail
-                = sentPacketDetails.remove(entry.getKey());
-
             if (packetDetail != null && videoStream != null)
```

Comment:
Am I missing something, or is videoStream useless in this method?

Reply:
No, it's not useless. For the outgoing RTCP TCC packets we need to set the source/media SSRC to be the primary SSRC of the incoming video media stream, otherwise Chrome might ignore the TCC packets.

Reply:
But this method handles incoming TCC, not outgoing. What am I missing?

Reply:
You're right.. I was thinking about something else. It looks like this is completely useless, thanks for noticing @bgrozev, I'll test it to make sure it doesn't break anything and remove it.